src/aiida/orm/nodes/data/code/installed.py

```diff
@@ -137,6 +137,10 @@ def validate_filepath_executable(self):
             with override_log_level():  # Temporarily suppress noisy logging
                 with self.computer.get_transport() as transport:
                     file_exists = transport.isfile(str(self.filepath_executable))
```

Comment:
if  `file_exists` is `False`, one can immediately raise here. Maybe no need to go for the next steps

Reply:
Yeah, the code was still incorrect there anyway, as `file_exists` needs to be True for the next lines ^^ This should be a draft PR. Just wanted to get it out yesterday already to see if the SSH testing issue also appears through GitHub actions. I will refactor this part. Thanks 🙏🏽

Reply:
Good point, I refactored that part!